### DISCUSSION TITLE
RPC call to stripe FDW returns empty result about 50% of the time

### DISCUSSION THREAD

**Discussion by vincenzon** [supabase/supabase#21065] [Questions]
I have an rpc call that queries stripe FDW tables (see below). When I call it from my webapp it returns an empty result `{data: [], error: null}` about half the time. The rest of the time it works as expected (2 rows). 

Any idea what I need to do to get it to work all the time? I thought it might be timing out, that's why I added the statement timeout, but that didn't help.

Thanks for any help.

```
create or replace function public.get_subscription_prices()
returns table (
    name text,
    product_id text,
    price_id text, 
    price real
)
language plpgsql security definer as
$$
#variable_conflict use_column
begin
    set statement_timeout to 60000;
    return QUERY
        select p.name, p.id as product_id, q.id as price_id, (q.unit_amount / 100.0)::real as price
        from stripe.products p
        left join stripe.prices q on p.id = q.product
        where p.active and q.active;
end
$$;
```

**Comment by GaryAustin1** [comment]
The #variable_conflict may be changing things based on the conflict.  I'm not that familiar with it, but it could certainly change the function operation from call to call.

**Comment by vincenzon** [comment]
That's part of my boilerplate for writing stored procedures. In this case it isn't needed so I removed it. No change, I still see about half the requests returning empty results. I tried varying the logged in user (multiple signed in users, and having no signed in user) and this didn't seem to matter either. 

It feels like something is timing out on the supabase <-> stripe connection, but that's just instinct. Is there any FDW specific logging?
